### ISSUE TITLE
Syntactically invalid R code produced by View() inspector icon for objects with invalid names

### ISSUE THREAD

**Issue by mattwarkentin** [rstudio/rstudio#6763]
Hi,

This is a pretty minor bug. If you have named an object with a syntactically invalid name wrapped in back-ticks (i.e. with spaces or unusual characters), when you open the object in the RStudio via `View()`, if you click the icon on the far right to inspect a sub-element, the R code pasted to the console is invalid due to the lacking back-ticks.

For example:
```r
`iris-data` <- as.list(iris)
View(`iris-data`)
```
If you click the _table+arrow_ icon you get:
```r
iris-data[["Sepal.Length"]]
# Error in data[["Sepal.Length"]] : 
#  object of type 'closure' is not subsettable
```
`iris-data[["Sepal.Length"]]` should be:
````
`iris-data`[["Sepal.Length"]]
````

I noticed this small bug when I ran a script as a local job (`Source as Local Job...`). I requested the  results from the job be returned via the option: `To results object in global environment`. Since I often name my R scripts with `-`, the returned object suffers from the same minor issue as above, as it is an invalid R object name.

This is an ultra minor bug, but thought I would share anyway.

**Comment by R-Hannibal** [comment]
Thanks so much for reporting this issue @mattwarkentin .

Following the very detailed steps provided I was able to reproduce this issue in RStudio 1.2.5033. However it looks the issue has been fixed in our upcoming 1.3 release (was not able to reproduce on 1.3.952). The latest 1.3 build is currently available on our preview page if you'd like to give it a try ([Preview Build](https://www.rstudio.com/products/rstudio/download/preview/)). 

Warning: preview builds are intended for testing and feedback purposes, and may therefore be less stable. We should see a stable release of 1.3 coming soon.
